Update manual deployment guide (#201)
scalar-labs/scalar-kubernetes: docs/ManualDeploymentGuideScalarDLOnEKS.md

> # Deploy ScalarDL Ledger on Amazon EKS (Amazon Elastic Kubernetes Service)

Reviewer: ```suggestion
# Deploy ScalarDL Ledger on Amazon Elastic Kubernetes Service (EKS)
```

> ScalarDL is scalable and practical Byzantine fault detection middleware for transactional database systems, which achieves correctness, scalability, and database agnosticism.
> 
> ScalarDL is composed of [Ledger](https://github.com/scalar-labs/scalardl/blob/master/docs/getting-started.md), [Auditor](https://github.com/scalar-labs/scalardl/blob/master/docs/getting-started-auditor.md), and [Client SDK](https://github.com/scalar-labs/scalardl/tree/master/docs#client-sdks). ScalarDL Ledger manages application data in its own unique way using hash-chain and digital signature. ScalarDL Auditor is an optional component and manages a copy of Ledger data without depending on Ledger to identify the discrepancy between Ledger and Auditor data. The Client SDK is a set of user-facing programs to interact with Ledger and Auditor.
> 
> We can deploy **ScalarDL Ledger** on any Kubernetes services. This document explains how to deploy **ScalarDL Ledger** on EKS.
> 
> ## What we create
> 
> In this guide, we create the following environment on your AWS account.

Reviewer: Similar to my other comment in the other deployment guides, I feel like users should already know what ScalarDL Ledger is at this point. How about if we simplify the beginning of this document as follows?:

```suggestion
This document explains how to deploy **ScalarDL Ledger** on Amazon Elastic Kubernetes Service (EKS).

In this guide, you will create the following environment in your AWS environment account.
```

![image](./images/png/EKS_ScalarDL_Ledger.drawio.png)

## Step 1. Subscribe to ScalarDL Ledger in AWS Marketplace

You can get the ScalarDL Ledger container image from [AWS Marketplace](https://aws.amazon.com/marketplace/pp/prodview-3jdwfmqonx7a2). First, you need to subscribe to it. For more details on how to subscribe to ScalarDL Ledger in AWS Marketplace, see [Subscribe to Scalar products from AWS Marketplace - How to install Scalar products through AWS Marketplace](./AwsMarketplaceGuide.md#subscribe-to-scalar-products-from-aws-marketplace).

## Step 2. Create an EKS cluster

> Create an EKS cluster for the deployment of ScalarDL Ledger. See [Create an Amazon EKS cluster for Scalar products](./CreateEKSClusterForScalarProducts.md) for more details.

Reviewer: ```suggestion
You must create an EKS cluster for the ScalarDL Ledger deployment. For details, see [Create an Amazon EKS cluster for Scalar products](./CreateEKSClusterForScalarProducts.md).
```

## Step 3. Set up a database for ScalarDL Ledger

> ScalarDL Ledger uses ScalarDB in its internal to access a database and ScalarDB supports [several databases](https://github.com/scalar-labs/scalardb/blob/master/docs/scalardb-supported-databases.md). You need to prepare a database before you deploy ScalarDL Ledger. See [Set up a database for ScalarDB/ScalarDL deployment on AWS](./SetupDatabaseForAWS.md) for more details.

Reviewer: ```suggestion
You must prepare a database before deploying ScalarDL Ledger. Because ScalarDL Ledger uses ScalarDB internally to access databases, refer to [ScalarDB Supported Databases](https://github.com/scalar-labs/scalardb/blob/master/docs/scalardb-supported-databases.md) to see which types of databases ScalarDB supports.

For details on setting up a database, see [Set up a database for ScalarDB/ScalarDL deployment on AWS](./SetupDatabaseForAWS.md).
```

## Step 4. Create a bastion server

> For executing some tools to deploy and manage ScalarDL Ledger on EKS, you need to prepare a bastion server in the same VPC of the EKS cluster you created in **Step 2**. See [Create a bastion server](./CreateBastionServer.md) for more details.

Reviewer: ```suggestion
To execute some tools for deploying and managing ScalarDL Ledger on EKS, you must prepare a bastion server in the same Amazon Virtual Private Cloud (VPC) of the EKS cluster you created in **Step 2**. For details, see [Create a Bastion Server](./CreateBastionServer.md).
```

> ## Step 5. Prepare a custom values file of Helm 

Reviewer: Do we need to configure two custom values files in this case (one for ScalarDL Ledger and one for ScalarDL Schema Loader (for Ledger)? If so, let's use the following:

```suggestion
## Step 5. Prepare custom values files for the Scalar Helm Charts for both ScalarDL Ledger and ScalarDL Schema Loader
```

If only one custom values file needs to be configured, let's use the following:

```suggestion
## Step 5. Prepare a custom values file for the Scalar Helm Chart
```

> You need to configure a custom values file for the Helm Chart of ScalarDL Ledger and ScalarDL Schema Loader based on your environment (e.g., access information of the database you created in **Step 3**). See [Configure a custom values file for Scalar Helm Charts](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-file.md) for more details.

Reviewer: The following assumes we need to configure two custom values files (one for ScalarDL Ledger and one for ScalarDL Schema Loader [for Ledger]).

```suggestion
To perform a task, like accessing information in the database that you created in **Step 3**, you must configure custom values files for the Scalar Helm Charts for both ScalarDL Ledger and ScalarDL Schema Loader (for Ledger) based on your environment. For details, see [Configure a custom values file for Scalar Helm Charts](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-file.md).
```

If only one custom values file needs to be configured, let's use the following:

```suggestion
To perform a task, like accessing information in the database that you created in **Step 3**, you must configure a custom values file for the Scalar Helm Chart for ScalarDL Ledger and ScalarDL Schema Loader (for Ledger) based on your environment. For details, see [Configure a custom values file for Scalar Helm Charts](https://github.com/scalar-labs/helm-charts/blob/main/docs/configure-custom-values-file.md).
```

> ## Step 6. Deploy ScalarDL Ledger using Scalar Helm Chart

Reviewer: ```suggestion
## Step 6. Deploy ScalarDL Ledger by using the Scalar Helm Chart
```

> Deploy ScalarDL Ledger on your EKS cluster using Scalar Helm Chart. See [Deploy Scalar products using Scalar Helm Charts](https://github.com/scalar-labs/helm-charts/blob/main/docs/how-to-deploy-scalar-products.md) for more details.

Reviewer: ```suggestion
You must deploy ScalarDL Ledger in your EKS cluster by using the Scalar Helm Chart. For details, see [Deploy Scalar products using Scalar Helm Charts](https://github.com/scalar-labs/helm-charts/blob/main/docs/how-to-deploy-scalar-products.md).
```

> Note: We recommend creating a dedicated namespace by using the `kubectl create ns scalardl-ledger` command and deploying ScalarDL Ledger in the namespace by using the `-n scalardl-ledger` option with the `helm install` command.

Reviewer: ```suggestion
**Note:** We recommend creating a dedicated namespace by using the `kubectl create ns scalardl-ledger` command and deploying ScalarDL Ledger in the namespace by using the `-n scalardl-ledger` option with the `helm install` command.
```

> ## Step 7. Check the status of ScalarDL Ledger deployment

Reviewer: ```suggestion
## Step 7. Check the status of your ScalarDL Ledger deployment
```

> After deploying ScalarDL Ledger on your EKS cluster, you need to check the status of each component. See [What you might want to check on a regular basis](./RegularCheck.md) for more details.

Reviewer: I updated the title of the linked doc below to be more specific by referencing the Kubernetes environment. If this title looks OK, let's update the title of the actual `RegularCheck.md` doc as well.

```suggestion
After deploying ScalarDL Ledger in your EKS cluster, you must check the status of each component. For details, see [Components to Regularly Check When Running in a Kubernetes Environment](./RegularCheck.md).
```

> ## Step 8. Monitoring for ScalarDL Ledger deployment

Reviewer: ```suggestion
## Step 8. Monitor your ScalarDL Ledger deployment
```

> After deploying ScalarDL Ledger on your EKS cluster, we recommend monitoring the deployed components and collecting their logs, especially in production. See [Monitoring Scalar products on a Kubernetes cluster](./K8sMonitorGuide.md) and [Collecting logs from Scalar products on a Kubernetes cluster](./K8sLogCollectionGuide.md) for more details.

Reviewer: ```suggestion
After deploying ScalarDL Ledger in your EKS cluster, we recommend monitoring the deployed components and collecting their logs, especially in production. For details, see [Monitoring Scalar products on a Kubernetes cluster](./K8sMonitorGuide.md) and [Collecting logs from Scalar products on a Kubernetes cluster](./K8sLogCollectionGuide.md).
```

> 
> ---

Reviewer: ```suggestion
```

> ## Uninstall ScalarDL Ledger on EKS

Reviewer: ```suggestion
## Remove ScalarDL Ledger from EKS
```

> If you want to uninstall the environment you created, please uninstall/remove resources in the reverse order of creation.

Reviewer: ```suggestion
If you want to remove the environment that you created, please remove all the resources in reverse order from which you created them in.
```
